sources/Outputs/src/ParGenerator.cpp

```diff
@@ -248,17 +260,12 @@ ParGenerator::writeGenerator(const algo::GeneratorDefinition& def, const std::st
   // Qmax and QMin are determined in dynawo according to reactive capabilities curves and min max
   // we need a small numerical tolerance in case the starting point of the reactive injection is exactly
   // on the limit of the reactive capability curve
-  if (def.hasTransformer() && (def.isUsingRectangularDiagram())) {
-    set->addParameter(helper::buildParameter("generator_QMin", def.qmin - 1));
-    set->addParameter(helper::buildParameter("generator_QMax", def.qmax + 1));
-  } else {
-    set->addParameter(helper::buildParameter("generator_QMin0", def.qmin - 1));
-    set->addParameter(helper::buildParameter("generator_QMax0", def.qmax + 1));
-  }
+  set->addParameter(helper::buildParameter("generator_QMin0", def.qmin - 1));
```

Comment:
I still quite don't get what we do here : if we have a diagram then we have QMin0/QMax0, if infinite limits or rectangular diagram the QMin/QMax, is that what's done here ?

Reply:
I have checked it in the test TestIIDM_launch_svc, and we write QMin0 and QMax0 for a gen with rectangular diagram, which is not as wanted (we already have QMin/QMax in the macro set)

Reply:
Now I dump those parameters only when the diagram exists and is not rectangular, is it ok?

Reply:
Yes, perfect